### PR TITLE
fix: verify the user from keycloak and sanitize emails

### PR DESCRIPTION
**Pull request by JivusAyrus**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Review by Aenimus — APPROVED** [review]
LGTM

File: controlplane/src/core/bufservices/PlatformService.ts [hunk 1]
```diff
@@ -3932,7 +3932,27 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
 
         await opts.keycloakClient.authenticateClient();
 
-        const user = await userRepo.byEmail(req.email);
+        const keycloakUser = await opts.keycloakClient.client.users.find({
+          max: 1,
+          email: req.email,
+          realm: opts.keycloakRealm,
+          exact: true,
```
Comment (StarpTech) [inline controlplane/src/core/bufservices/PlatformService.ts:3939]: Will this also work when someone use a capitalized email?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: controlplane/src/core/bufservices/PlatformService.ts (hunk 1 above)
Comment (JivusAyrus) [reply controlplane/src/core/bufservices/PlatformService.ts:3939]: yes

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**